sdk/core/azure_core/src/policies/bearer_token_policy.rs

```diff
@@ -22,6 +27,35 @@ impl BearerTokenCredentialPolicy {
         Self {
             credential,
             scopes: scopes.into_iter().map(|s| s.into()).collect(),
+            access_token: Arc::new(RwLock::new(None)),
+        }
+    }
+
+    fn scopes(&self) -> Vec<&str> {
+        self.scopes
+            .iter()
+            .map(String::as_str)
+            .collect::<Vec<&str>>()
+    }
+
+    fn refresh_token(&self, new_access_token: &AccessToken) {
+        let mut access_token = self.access_token.write().unwrap();
```

Comment:
Going to start a comment here but this serves as discussion for all uses of `unwrap()` that still exists here.
I know we have alternatives such as `unwrap_or_default()` but I couldn't quite wrap my head around how we would use it in these instances.

One thing is that these `unwrap()` should (from my understanding) never really raise since calling `.write()` or `.read()` before blocks the current thread until the lock can be acquired, and `unwrap()` should only raise if it's poisoned (a writer thread panics while holding an exclusive lock) which given the current code shouldn't ever occur? 

As always, open to feedbacks on how to make this more rigorous and robust! Thanks 😄